src/components/TodosContext.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { ChangeEvent, useState, KeyboardEvent } from 'react';
+import { Todo } from '../types/Todo';
+import { Status } from '../services/Status';
+import { useLocalStorage } from '../services/localStorage';
+import { editTodoFunction } from '../types/EditTodoFunction';
+
+type TodoContext = {
+  todos: Todo[];
+  setTodos: (value: Todo[]) => void;
+  toggleAll: boolean;
+  setToggleAll: (status: boolean) => void;
+  filter: Status;
+  setFilter: React.Dispatch<React.SetStateAction<Status>>;
+  filteredTodos: Todo[];
+  handleInputChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  handleAddTodo: () => void;
+  handleKeyDown: (event: React.KeyboardEvent<HTMLInputElement>) => void;
+  deleteTodo: (id: number) => void;
+  clearForm: () => void;
+  toggleCompleted: (id: number) => void;
+  toggleAllChange: () => void;
+  editTodo: editTodoFunction;
+  newTodos: string;
+  setNewTodos: React.Dispatch<React.SetStateAction<string>>;
+};
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const TodosContext = React.createContext<TodoContext>({
+  todos: [],
+  setTodos: () => {},
+  toggleAll: false,
+  setToggleAll: () => {},
+  filter: Status.ALL,
+  setFilter: () => {},
+  filteredTodos: [],
+  handleInputChange: () => {},
+  handleAddTodo: () => {},
+  handleKeyDown: () => {},
+  deleteTodo: () => {},
+  clearForm: () => {},
+  toggleCompleted: () => {},
+  toggleAllChange: () => {},
+  editTodo: () => {},
+  newTodos: '',
+  setNewTodos: () => {},
+});
+
+export const TodosProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useLocalStorage('todos', [] as Todo[]);
+  const [filter, setFilter] = useState(Status.ALL);
+  const [toggleAll, setToggleAll] = useState(false);
+  const [newTodos, setNewTodos] = useState('');
+
+  // відфільтрувати завдання відповідно до вибраного фільтра
+  const filteredTodos = todos.filter(todo => {
```

Comment:
The todos are being filtered on each render which can be a performance issue if the list is large. Consider filtering todos only when it's necessary (when filter or todos change).